src/unitxt/task.py

```diff
@@ -10,42 +13,141 @@ class Tasker:
         "inputs" whose value is a sub-dictionary of the input instance, consisting of all the fields listed in Arg 'inputs'.
         "outputs" -- for the fields listed in Arg "outputs".
         "metrics" -- to contain the value of Arg 'metrics'
-
     """
 
-    inputs: List[str]
-    outputs: List[str]
+    inputs: Union[Dict[str, Any], List[str]]
+    outputs: Union[Dict[str, Any], List[str]]
     metrics: List[str]
+    prediction_type: Optional[Any] = None
     augmentable_inputs: List[str] = []
 
+    @staticmethod
+    def process_type(data_type: Any) -> Any:
+        if isinstance(data_type, str):
+            data_type = data_type.replace("typing.", "")
+            return parse_type_string(data_type)
+        return data_type
+
     def verify(self):
+        for io_type in ["inputs", "outputs"]:
+            data = self.inputs if io_type == "inputs" else self.outputs
+            if not isinstance(data, Dict):
+                get_logger().warning(
+                    f"'{io_type}' should be a dict of field names and their types, instead only "
+                    f"'{data}' was passed. All types will be assumed to be 'Any'. In future version "
+                    f"of unitxt this will raise an exception."
+                )
+                data = {key: Any for key in data}
+                if io_type == "inputs":
+                    self.inputs = data
+                else:
+                    self.outputs = data
+
+        if not self.prediction_type:
+            get_logger().warning(
+                "'prediction_type' was not passed. It must be compatible with used metrics. "
+                "Trying to infer `prediction_type` based on specified metrics. In future "
+                "version of unitxt this will raise an exception."
+            )
+            sample_metric = fetch_artifact(self.metrics[0])[0]
+            self.prediction_type = sample_metric.get_prediction_type()
+
         for augmentable_input in self.augmentable_inputs:
             assert (
                 augmentable_input in self.inputs
             ), f"augmentable_input f{augmentable_input} is not part of {self.inputs}"
 
-    def process(
-        self, instance: Dict[str, Any], stream_name: Optional[str] = None
-    ) -> Dict[str, Any]:
-        try:
-            inputs = {key: instance[key] for key in self.inputs}
-        except KeyError as e:
-            raise KeyError(
-                f"Unexpected FormTask input column names ({[key for key in self.inputs if key not in instance]})."
-                f"The available input names: {list(instance.keys())}"
-            ) from e
+    def check_metrics_type(self) -> None:
+        for metric_name in self.metrics:
+            metric = fetch_artifact(metric_name)[0]
+            metric_prediction_type = metric.get_prediction_type()
+
+            if self.prediction_type == metric_prediction_type:
+                continue
+
+            prediction_type = self.process_type(self.prediction_type)
```

Comment:
instead of process_type.  Let's create get_prediction_type() in the Task, which returns the type after parsing.  And do all comparison between the types.